docs/order-confirmation/index.md

```diff
@@ -0,0 +1,333 @@
+to render a table containing all items in an order.
+
+## `Order` component
+
+Create a new file `src/order-confirmation/Order.re` and add the following code:
+
+<<< Snippets.re#order
+
+There's a lot going on here:
+
+- The primary type of `Order` is `array(Item.t)`, which is an array of
+  variants.
+- The `make` function has a single labeled argument, `~items`, of type `t`.
+  This means the `Order` component has a single prop named `items`.
+- We sum up the prices of all items using
+  [Js.Array.reduce](https://melange.re/v2.1.0/api/re/melange/Js/Array/index.html#val-reduce).
+  This is identical to JavaScript's [Array.reduce
+  method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce),
+  except that the Melange version requires the initial value to be passed in.
+- For each order, we render an `Item` component via
+  [Js.Array.map](https://melange.re/v2.1.0/api/re/melange/Js/Array/index.html#val-map),
+  which is equivalent to the [Array.map
+  method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).
+- The `React.array` function has the type signature `array(React.element) =>
+  React.element`, meaning it converts an array of `React.element`s to
+  `React.element`. All expressions in JSX must have the type `React.element`.
+
+## `Index.re`
+
+Render the `Order` component inside `Index.re`:
+
+<<< Snippets.re#index
+
+Run `make serve` inside `src/order-confirmation` to see your new app in action.
+Open your browser's dev console, where you should see a warning:
+
+```
+Warning: Each child in a list should have a unique "key" prop.
+```
+
+Oops, we should've used
+[Js.Array.mapi](https://melange.re/v2.1.0/api/re/melange/Js/Array/index.html#val-mapi)[^3]
+instead so we could set the `key` prop for each `Item` component:
+
+<<< Snippets.re#mapi
+
+You've got a basic component now, but it looks... not so great.[^4]
+
+## Style with CSS
+
+Let's add styling, but instead of using the `style` prop as we did in previous
+chapters, let's use good old CSS. First, install `style-loader` and `css-loader`
+webpack plugins:
+
+```
+npm install --save-dev css-loader style-loader
+```
+
+Add a new rule to `webpack.config.js` so that you can import .css files as
+modules:
+
+<<< @/../webpack.config.js{5-12}
+
+Add a new file
+`src/order-confirmation/item.css` and give it these styles:
+
+<<< @/../src/order-confirmation/item.css
+
+## `mel.raw` extension node
+
+In OCaml, there is no syntax to import from files, because all modules within a
+project are visible to all other modules[^5]. However, we can make use of
+JavaScript's `import` syntax by using the [mel.raw extension
+node](https://melange.re/v2.1.0/communicate-with-javascript/#generate-raw-javascript),
+which allows us to embed raw JavaScript in our OCaml code. Add the following
+code to the top of `Item.re`:
+
+```reason
+[%%mel.raw {|import "./item.css"|}];
```

Comment:
This is minor, but I wonder if it'd be better to use `module`?

```suggestion
[@module "./item.css"] external css: string = "default";
ignore(css);
```

`mel.raw` with `import` is simpler, but I'm not sure if it could break if the module system changes from `es6` to `commonjs`.

Reply:
Who knows. I've seen much worse 😄 in any case, not a big deal. Can stay as is.

Reply:
Since I'm moving all the styling stuff to the next chapter, I'll add the `@mel.module` approach there